scaleway/table_scaleway_baremetal_server.go

```diff
@@ -0,0 +1,279 @@
+	zone := plugin.GetMatrixItem(ctx)["zone"].(string)
+
+	parseZoneData, err := scw.ParseZone(zone)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_baremetal_server.listBaremetalServers", "zone_parsing_error", err)
+		return nil, err
+	}
+
+	quals := d.KeyColumnQuals
+	if quals["zone"] != nil && quals["zone"].GetStringValue() != zone {
+		return nil, nil
+	}
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_baremetal_server.listBaremetalServers", "connection_error", err)
+		return nil, err
+	}
+
+	// Create SDK objects for Scaleway Instance product
+	baremetalApi := baremetal.NewAPI(client)
+
+	req := &baremetal.ListServersRequest{
+		Zone: parseZoneData,
+		Page: scw.Int32Ptr(1),
+	}
+	// Additional filters
+	if quals["name"] != nil {
+		req.Name = scw.StringPtr(quals["name"].GetStringValue())
+	}
+	// Retrieve the list of servers
+	maxResult := int64(100)
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < maxResult {
+			maxResult = *limit
+		}
+	}
+	req.PageSize = scw.Uint32Ptr(uint32(maxResult))
+
+	var count int
+
+	for {
+		resp, err := baremetalApi.ListServers(req)
+		if err != nil {
+			plugin.Logger(ctx).Error("scaleway_baremetal_server.listBaremetalServers", "query_error", err)
+			return nil, err
+		}
+
+		for _, baremetal := range resp.Servers {
+			d.StreamListItem(ctx, baremetal)
+
+			// Increase the resource count by 1
+			count++
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if resp.TotalCount == uint32(count) {
+			break
+		}
+		req.Page = scw.Int32Ptr(*req.Page + 1)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getBaremetalServer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	zone := plugin.GetMatrixItem(ctx)["zone"].(string)
+
+	parseZoneData, err := scw.ParseZone(zone)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_baremetal_server.listBaremetalServers", "zone_parsing_error", err)
+		return nil, err
+	}
+
+	if d.KeyColumnQuals["zone"].GetStringValue() != zone {
+		return nil, nil
+	}
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_baremetal_server.getBaremetalServer", "connection_error", err)
+		return nil, err
+	}
+
+	// Create SDK objects for Scaleway Baremetal product
+	baremetalApi := baremetal.NewAPI(client)
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+	baremetalZone := d.KeyColumnQuals["zone"].GetStringValue()
+
+	// No inputs
+	if id == "" && baremetalZone == "" {
+		return nil, nil
+	}
+
+	data, err := baremetalApi.GetServer(&baremetal.GetServerRequest{
+		ServerID: id,
+		Zone:     parseZoneData,
```

Comment:
```suggestion
		Zone:     baremetalZone,
```

Shouldn't this be `baremetalZone` instead?

Reply:
I'm not sure :/ i prefer call the api with the same zone that the connection [instanced](https://github.com/turbot/steampipe-plugin-scaleway/pull/17/files#diff-85fe56f123e32901ccd3de88e08109c528bcfba72e47ffd046ab87a065a67dbbR238).  

is based on file [instance_server.go](https://github.com/turbot/steampipe-plugin-scaleway/blob/main/scaleway/table_scaleway_instance_server.go)

Reply:
I agree with @jplanckeel , I think keeping it as `parseZoneData` is nice since it's consistent with what we do in the instance server table.